tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/OdsOrganisationSteps.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using IdentityModel.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps.Common;
+using NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Utils;
+using TechTalk.SpecFlow;
+using TechTalk.SpecFlow.Assist;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
+{
+    [Binding]
+    public sealed class OdsOrganisationSteps
+    {
+        private readonly ScenarioContext _context;
+        private readonly Response _response;
+        private readonly OdsApiSteps _api;
+
+        private readonly string _organisationUrl;
+
+        public OdsOrganisationSteps(Response response, ScenarioContext context, Settings settings, OdsApiSteps api)
+        {
+            _response = response;
+            _context = context;
+            _api = api;
+            _organisationUrl = settings.OrganisationApiBaseUrl + "/api/v1/ods";
+        }
+
+        [Given(@"Ods Organisations exist")]
+        public async Task GivenOrganisationsExist(Table table)
+        {
+            var odsOrganisations = table.CreateSet<OdsApiResponseTable>();
+
+            foreach (OdsApiResponseTable org in odsOrganisations)
+            {
+                var odsApiOrganisation = TransformIntoOdsApiFormat(org);
+                var odsApiOrganisationAsJson = JsonConvert.SerializeObject(odsApiOrganisation);
+                await _api.SetUpGETEndpoint(org.OdsCode, odsApiOrganisationAsJson);
+            }
+        }
+
+        [When(@"a GET request is made for an Ods organisation with code (.*)")]
+        public async Task WhenAGETRequestIsMadeForAnOdsOrganisationWithOdsCode(string odsCode)
+        {
+            using var client = new HttpClient();
+            client.SetBearerToken(_context.Get(ScenarioContextKeys.AccessToken, ""));
+            _response.Result = await client.GetAsync(new Uri($"{_organisationUrl}/{odsCode}"));
```

Comment:
Would it simplify the API calls if we used Flurl?

Reply:
To compare

Now:

```    
      using var client = new HttpClient();
      client.SetBearerToken(_context.Get(ScenarioContextKeys.AccessToken, ""));
      _response.Result = await client.GetAsync(new Uri($"{_organisationUrl}/{odsCode}"));
```
With Flurl:
```
            _response.Result = await _organisationUrl
                .AppendPathSegment(odsCode)
                .WithOAuthBearerToken(_context.Get(ScenarioContextKeys.AccessToken, ""))
                .AllowAnyHttpStatus()
                .GetAsync();
```

Now, I quite like the fluent syntax, so they would look more pleasing to my eyes.

However, there are some cons
- we need to bring in another dependency 
- we'd probably need to refactor all the other operations using httpclient as to remain consistent (consistency rules)

Looking at the thing objectively, it will not be worth the refactor at the moment solely because I like it, however if more people agree that using flurl here is good, we can raise a task to refactor the tests (which shouldn't take that long).

TL;DR I'd like to leave it as is for now, discuss this with other devs and if they're in favour of refactoring, we refactor all the tests to use Flurl.

Thoughts? @bjss-chris

Reply:
That sounds sensible to me @assoomro and thanks for preparing the example.

Reply:
There's only one other tiny thing I'll mention here. I tend to prefer `string.Empty` over it's literal equivalent. The reason I prefer it is because it means I have no doubt about intent (i.e. the empty string isn't a mistake).

However, we don't currently have any standards covering this as far as I'm aware and as there is no functional difference I'll leave the decision up to you.

Reply:
Good spot here, I agree that string.Empty would be better in this case. We can bring it up in today's code standards meeting along with the flurl proposition and hopefully raise TD tasks to clear this up (as it's used in all over the place).